src/ServerStream.php

```diff
@@ -31,28 +31,40 @@
  */
 namespace Google\ApiCore;
 
+use Google\Auth\Logging\LoggingTrait;
+use Google\Auth\Logging\RpcLogEvent;
+use Google\Protobuf\Internal\Message;
 use Google\Rpc\Code;
+use Psr\Log\LoggerInterface;
 
 /**
  * ServerStream is the response object from a server streaming API call.
  */
 class ServerStream
 {
+    use LoggingTrait;
+
     private $call;
     private $resourcesGetMethod;
+    private null|LoggerInterface $logger;
 
     /**
      * ServerStream constructor.
      *
      * @param ServerStreamingCallInterface $serverStreamingCall The server streaming call object
      * @param array $streamingDescriptor
+     * @param null|LoggerInterface $logger A PSR-3 compliant logger.
      */
-    public function __construct($serverStreamingCall, array $streamingDescriptor = [])
-    {
+    public function __construct(
+        $serverStreamingCall,
+        array $streamingDescriptor = [],
+        null|LoggerInterface $logger = null
```

Comment:
can take advantage of declaring properties in php
```suggestion
        private null|LoggerInterface $logger = null
```